lxd/storage_zfs.go

```diff
@@ -547,7 +547,91 @@ func (s *storageZfs) StoragePoolUpdate(writable *api.StoragePoolPut, changedConf
 	}
 
 	if shared.StringInSlice("volume.size", changedConfig) {
-		return fmt.Errorf("the \"volume.size\" property cannot be changed")
+		volumes, err := db.StoragePoolVolumesGet(s.s.DB, s.poolID,
+			[]int{storagePoolVolumeTypeContainer,
+				storagePoolVolumeTypeCustom})
+		if err != nil {
+			if err != db.NoSuchObjectError {
+				return err
+			}
+		}
+
+		poolName := s.getOnDiskPoolName()
+		size := int64(0)
+		if writable.Config["volume.size"] != "" {
+			size, err = shared.ParseByteSizeString(writable.Config["volume.size"])
+			if err != nil {
+				return err
+			}
+		}
+
+		for _, volume := range volumes {
+			var data interface{}
+			var volumeType int
+
+			// Check if the storage volume itself already has a
+			// local "size" property set.
+			if volume.Config["size"] != "" {
+				continue
+			}
```

Comment:
So hmm, I may be getting confused, but isn't the design that volume.size on the storage pool becomes the size value for any newly created storage volume?
That is, once a volume has been created its size won't be modified unless the user directly goes and modifies it.

If that's the case, then none of this code is actually needed as a change to "volume.size" will not affect any existing volumes, only newly created ones.